src/main/java/com/PBA/authservice/facade/UserFacadeImpl.java

```diff
@@ -46,4 +50,21 @@ public UserDto getUser(UUID uid) {
         UserProfileDto userProfileDto = activeUserMapper.toUserProfileDto(activeUserProfile);
         return activeUserMapper.toUserDto(activeUser, userProfileDto);
     }
+
+    @Override
+    public UserDto verifyUser(UUID validationCode) {
+        PendingUser userToValidate = pendingUserService.getPendingUserByValidationCode(validationCode);
+        userValidator.validatePendingUser(userToValidate);
```

Comment:
```suggestion
        if (userToValidate.isExpired()) {
           throw new UserNotFoundException("...");
        }
```

To check for expired use the PendingUser object to have business logic.